react/__tests__/LocationInput.test.tsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react'
+import { render, screen, fireEvent, waitFor } from '@vtex/test-tools/react'
+import { MockedResponse } from '@apollo/react-testing'
+import { Address } from 'vtex.checkout-graphql'
+import { AddressRules, CountryRules, Field } from 'vtex.address-context/types'
+import { AddressContextProvider } from 'vtex.address-context/AddressContext'
+
+import LocationInput from '../LocationInput'
+import { validPostalCode } from '../__fixtures__/graphql/getAddressFromPostalCode.fixture'
+
+interface SetupProps {
+  address?: Address | null
+  countries?: string[]
+  rules?: AddressRules
+  graphqlMocks?: MockedResponse[]
+}
+
+interface RenderComponentProps extends SetupProps {
+  locationInputProps?: React.ComponentProps<typeof LocationInput>
+}
+
+const defaultGraphqlMocks: MockedResponse[] = []
+
+describe('LocationInput', () => {
+  const renderComponent = ({
+    locationInputProps = {},
+    address = { country: 'BRA' },
+    countries = [],
+    rules = {
+      BRA: {
+        fields: {
+          postalCode: {
+            mask: '99999-999',
+            pattern: '^(\\d){5}-?(\\d){3}$',
+            additionalData: {
+              forgottenURL:
+                'https://buscacepinter.correios.com.br/app/endereco/',
+            },
+          } as Field,
+        },
+      } as CountryRules,
+    },
+    graphqlMocks = [],
+  }: RenderComponentProps) => {
+    return render(
+      <AddressContextProvider
+        address={address}
+        countries={countries}
+        rules={rules}
+      >
+        <LocationInput {...locationInputProps} />
+      </AddressContextProvider>,
+      {
+        graphql: {
+          mocks: [...defaultGraphqlMocks, ...graphqlMocks],
+          addTypename: false,
+        },
+      }
+    )
+  }
+
+  afterEach(() => {
+    jest.resetAllMocks()
+  })
+
+  it('should format and submit a valid postal code', async () => {
+    const mockedOnSuccess = jest.fn()
+
+    renderComponent({
+      locationInputProps: { onSuccess: mockedOnSuccess },
+      graphqlMocks: [validPostalCode],
+    })
+
+    const input = screen.getByRole('textbox', {
+      name: /postal code/i,
+    })
+
+    input.focus()
+    fireEvent.change(input, { target: { value: '22250040' } })
+
+    const button = screen.getByRole('button')
+
+    // formatting happens on input blur, so this should be simulated manually
+    // see https://testing-library.com/docs/guide-events/#interactions-vs-events
+    button.focus()
```

Comment:
same as above, but you could explicitly blur the input here instead of focusing on the button, or use the [`user-event`](https://github.com/testing-library/user-event) library

Reply:
this lib looks pretty good. I'll try it!